src/fsharp/fsi/fsi.fs

```diff
@@ -1641,18 +1663,124 @@ module internal MagicAssemblyResolution =
                stopProcessingRecovery e range0
                null
 
+#if NETSTANDARD
+        let probingFileNames (name: string) =
+            let isRooted = Path.IsPathRooted name
+            let useSuffix s = not (name.Contains(s + ".") || name.EndsWith(s))      // linux devs often append version # to libraries I.e mydll.so.5.3.2
+            let usePrefix = not (name.Contains(@"\") || name.Contains(@"/"))        // If name has directory information no add no prefix
+            let prefix = [| "lib" |]
+            let suffix = [|
+                    if RuntimeInformation.IsOSPlatform(OSPlatform.Windows) then
+                        ".dll"
+                        ".exe"
+                    elif RuntimeInformation.IsOSPlatform(OSPlatform.OSX) then
+                        ".so"
+                    else
+                        ".dylib"
+                |]
+
+            [|
+                yield name                                                                              // Bare name
+                if not (isRooted) then
+                    for s in suffix do
+                        if useSuffix s then                                                             // Suffix without prefix
+                            yield (sprintf "%s%s" name s)
```

Comment:
Nit, but ultimately I don't really care: just string concat instead of `sprintf`.

Reply:
I like it this way.